### PR TITLE
Blazor Hot Reload 7.0 improvements

### LINKED ISSUE #27560: Blazor Hot Reload 7.0 improvements
For the 7.0 Blazor WASM section of the doc ...

* New types
* Nested classes
* Static and instance methods to existing types
* Static fields and methods to existing types
* Static lambdas to existing methods
* Lambdas that capture `this` to existing methods that already captured `this` previously

Also, it currently reads 'support for' then goes on to mention some items that aren't supported. *Clean up that lingo!*

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 41ab3e64-1abc-987b-13db-4e49f2d68ff3
* Version Independent ID: 41ab3e64-1abc-987b-13db-4e49f2d68ff3
* Content: [.NET Hot Reload support for ASP.NET Core](https://learn.microsoft.com/en-us/aspnet/core/test/hot-reload?view=aspnetcore-7.0)
* Content Source: [aspnetcore/test/hot-reload.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/test/hot-reload.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-test**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #27560

This includes some lingo improvements.

* Are the lists correct?
* Is the phrasing good?

### CODE REVIEW

File: aspnetcore/test/hot-reload.md [hunk 1]
```diff
@@ -26,13 +26,45 @@ For more information on supported scenarios, see [Supported code changes (C# and
 
 ## Blazor WebAssembly
 
-Blazor WebAssembly Hot Reload support has the following conditions:
+::: moniker range=">= aspnetcore-7.0"
 
-* Hot Reload reacts to most changes to method bodies, such as adding, removing, and editing variables, expressions, and statements.
```
Comment (danroth27) [inline aspnetcore/test/hot-reload.md:31]: The first two bullet points in this list are still valid. The list you added below is for the edits that we *added* support for in .NET 7.

File: aspnetcore/test/hot-reload.md [hunk 2]
```diff
@@ -26,13 +26,45 @@ For more information on supported scenarios, see [Supported code changes (C# and
...
+* New types.
+* Nested classes.
+* Most changes to method bodies, such as adding, removing, and editing variables, expressions, and statements.
+* Changes to the bodies of [lambda expressions](/dotnet/csharp/language-reference/operators/lambda-expressions) and [local functions](/dotnet/csharp/programming-guide/classes-and-structs/local-functions).
+* Adding static and instance methods to existing types.
+* Adding static fields and methods to existing types.
+* Adding static lambdas to existing methods.
+* Adding lambdas that capture `this` to existing methods that already captured `this` previously.
+
+Blazor WebAssembly and Blazor Server Hot Reload have support for when components reset their parameters to their default values when a value is removed.
```
Comment (danroth27) [inline aspnetcore/test/hot-reload.md:42]:
```suggestion
Note that when an attribute is removed that previously set the value of a component parameter, the component parameter is set back to its default value.
```

@MackinnonBuck What do we use as the "default" value? Do we just do `default<T>` or do we look at any default values for the property?

**Comment by danroth27** [comment]
We need @lambdageek 's signoff on this.

File: aspnetcore/test/hot-reload.md [hunk 3]
```diff
@@ -26,13 +26,45 @@ For more information on supported scenarios, see [Supported code changes (C# and
...
+
+The following code changes aren't supported for Blazor WebAssembly apps:
+
+* Adding local functions.
```
Comment (lambdageek) [inline aspnetcore/test/hot-reload.md:46]:
```suggestion
```

File: aspnetcore/test/hot-reload.md [hunk 4]
```diff
@@ -26,13 +26,45 @@ For more information on supported scenarios, see [Supported code changes (C# and
...
+* Adding a new [`await` operator](/dotnet/csharp/language-reference/operators/await) or [`yield` keyword](/dotnet/csharp/language-reference/keywords/yield) expression.
+* Changing the names of method parameters.
+* Changes outside of method bodies.
```
Comment (lambdageek) [inline aspnetcore/test/hot-reload.md:49]:
```suggestion
* Adding new non-`static` fields, events or properties 
```

**Review by lambdageek — APPROVED** [review]
LGTM, but local functions are fine now, and changes outside of classes that add `static` members are fine. Only non-static fields/methods/properties are problematic.

File: aspnetcore/test/hot-reload.md (hunk 2 above)
Comment (MackinnonBuck) [reply aspnetcore/test/hot-reload.md]: @danroth27 What we do is if we detect that a component had one or more parameters removed in a hot reload edit, we completely destroy and re-create the component. This allows any custom component initialization logic to set missing parameters to their intended "default" values.

Comment (MackinnonBuck) [reply aspnetcore/test/hot-reload.md]: So in other words, the "default" value is the value that the parameter would have if it were unspecified.

File: aspnetcore/test/hot-reload.md [hunk 5]
```diff
@@ -26,13 +26,44 @@ For more information on supported scenarios, see [Supported code changes (C# and
...
+* New types.
+* Nested classes.
+* Most changes to method bodies, such as adding, removing, and editing variables, expressions, and statements.
+* Changes to the bodies of [lambda expressions](/dotnet/csharp/language-reference/operators/lambda-expressions) and [local functions](/dotnet/csharp/programming-guide/classes-and-structs/local-functions).
+* Adding static and instance methods to existing types.
+* Adding static fields and methods to existing types.
+* Adding static lambdas to existing methods.
+* Adding lambdas that capture `this` to existing methods that already captured `this` previously.
+
+Note that when an attribute is removed that previously set the value of a component parameter, the component parameter is set back to its default value.
```
Comment (MackinnonBuck) [inline aspnetcore/test/hot-reload.md:42]:
```suggestion
Note that when an attribute is removed that previously set the value of a component parameter, the component is disposed and re-initialized to set the removed parameter back to its default value.
```

**Review by danroth27 — APPROVED** [review]
LGTM

**Pull request merged**